cub/detail/exec_check_disable.cuh

```diff
@@ -0,0 +1,41 @@
+/*
+*  Copyright 2021 NVIDIA Corporation
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*/
+
+#pragma once
+
+#include <cub/util_compiler.cuh>
+
+/**
+ * \def CUB_EXEC_CHECK_DISABLE
+ * Wrapper around `#pragma nv_exec_check_disable`.
+ */
+
+// #pragma nv_exec_check_disable is only recognized by NVCC.
+#if defined(__CUDACC__) && \
+    !defined(_NVHPC_CUDA) && \
```

Comment:
As I understand `_NVHPC_CUDA` is a replacement for `__NVCOMPILER_CUDA__`. Since CUB uses both, it's a bit confusing. Are you going to replace `__NVCOMPILER_CUDA__` in the following PRs?

Reply:
`_NVHPC_CUDA` is preferred in the future -- the other one will not be around much longer AIUI